contracts/protocol/libraries/logic/SupplyLogic.sol

```diff
@@ -429,13 +428,26 @@ library SupplyLogic {
         DataTypes.ReserveData storage reserve,
         DataTypes.ExecuteWithdrawERC721Params memory params
     ) internal returns (uint64, uint64) {
+        uint256 amount = 0;
+        INToken nToken = INToken(xTokenAddress);
+        if (nToken.getXTokenType() == XTokenType.NTokenUniswapV3) {
+            uint256 tokenIdLength = params.tokenIds.length;
+            for (uint256 index = 0; index < tokenIdLength; index++) {
+                amount += IPriceOracleGetter(params.oracle).getTokenPrice(
+                    params.asset,
+                    params.tokenIds[index]
+                );
+            }
+        } else {
+            amount = params.tokenIds.length;
+        }
         DataTypes.TimeLockParams memory timeLockParams = GenericLogic
             .calculateTimeLockParams(
                 reserve,
                 DataTypes.TimeLockFactorParams({
                     assetType: DataTypes.AssetType.ERC721,
```

Comment:
Are we using ERC721 assetType when burning UniswapV3 LP token?
For other tokens, it's the count, and it's dollor value for UniswapV3 ?

Reply:
Actually it's ETH value.